internal/validation/validation.go

```diff
@@ -133,3 +134,26 @@ func ValidatePending(c *contracts.Contract, pBalance *uint64, pNonce *uint64) er
 	(*pNonce)++
 	return nil
 }
+
+// Validate takes in expected version, height, previousHash, and timeStamp
+// and compares them with the block's
+func ValidateBlock(b block.Block, version uint16, height uint64, previousHash []byte, timeStamp int64) bool {
+	// Check Version
+	if b.Version != version {
+		return false
+	}
+	// Check Height
+	if b.Height != height {
+		return false
+	}
+	// Check Previous Hash
+	if !(bytes.Equal(b.PreviousHash, previousHash)) {
+		return false
+	}
+	// Check timestamp
+	if b.Timestamp != timeStamp {
```

Comment:
This doesn't allow for timestamps that have been mined between "now" and the parameter timestamp.